_search-plugins/collapse-search.md

```diff
@@ -0,0 +1,231 @@
+        "_score": null,
+        "_source": {
+          "item": "Chocolate Cake",
+          "category": "cakes",
+          "price": 15,
+          "baked_date": "2023-07-01T00:00:00Z",
+          "baker": "Baker A"
+        },
+        "fields": {
+          "item": [
+            "Chocolate Cake"
+          ]
+        },
+        "sort": [
+          15
+        ]
+      }
+    ]
+  }
+}
+```
+
+The collapsed search results example will show only one "Chocolate Cake" entry, demonstrating how collapsing works in reducing redundancy.
+
+Collapsing affects only the top search results and does not change any aggregation results. The total number of hits shown in the response reflects all matching documents before collapsing is applied, including duplicates. However, the response doesn't tell you the exact number of unique groups formed by collapsing.
+
+---
+
+## Expanding collapsed results
+
+You can expand each collapsed top hit with the `inner_hits` property. 
+
+The following example request applies `inner_hits` to retrieve the lowest priced and most recent items for each type of cake.
+
+```json
+GET /bakery-items/_search
+{
+  "query": {
+    "match": {
+      "category": "cakes"
+    }
+  },
+  "collapse": {
+    "field": "item",
+    "inner_hits": [
+      {
+        "name": "cheapest_items",
+        "size": 1,
+        "sort": ["price"]
+      },
+      {
+        "name": "newest_items",
+        "size": 1,
+        "sort": [{ "baked_date": "desc" }]
+      }
+    ]
+  },
+  "sort": ["price"]
+}
+
+```
+
+### Multiple inner hits for each collapsed hit
+
+To obtain several groups of inner hits for each collapsed result, you can set different criteria for each group. For example, you could request the three lowest priced items and the three most recent items for every bakery item.
+
+```json
+GET /bakery-items/_search
+{
+  "query": {
+    "match": {
+      "category": "cakes"
+    }
+  },
+  "collapse": {
+    "field": "item",
+    "inner_hits": [
+      {
+        "name": "cheapest_items",
+        "size": 1,
+        "sort": ["price"]
+      },
+      {
+        "name": "newest_items",
+        "size": 3,
+        "sort": [{ "baked_date": "desc" }]
+      }
+    ]
+  },
+  "sort": ["price"]
+}
+
+
+```
+This query searches for documents in the `cakes` category and groups the search results by the `item_name` field. For each `item_name`, it retrieves the top three lowest priced items and the top three most recent items, sorted by `baked_date` in descending order.
+
+Expanding the groups is done by sending an additional query for each inner hit request for each collapsed hit returned in the response. This can significantly slow down the process if there are too many groups or inner hit requests. The `max_concurrent_group_searches` request parameter can be used to control the maximum number of concurrent searches allowed in this phase. The default is based on the number of data nodes and the default search thread pool size.
```

Comment:
```suggestion
You can expand the groups by sending an additional query for each inner hit request for each collapsed hit returned in the response. This can significantly slow down the process if there are too many groups or inner hit requests. The `max_concurrent_group_searches` request parameter can be used to control the maximum number of concurrent searches allowed in this phase. The default is based on the number of data nodes and the default search thread pool size.
```